runtime/lua/vim/_meta/api.lua

```diff
@@ -744,6 +768,12 @@ function vim.api.nvim_buf_set_option(buffer, name, value) end
 --- `start_row = end_row = row` and `start_col = end_col = col`. To delete the
 --- text in a range, use `replacement = {}`.
 ---
+--- Note:
+--- • Prefer `nvim_buf_set_lines()` (for performance) to add or delete entire
+---   lines.
+--- • Prefer `nvim_paste()` or `nvim_put()` to insert (instead of replace)
+---   text at cursor.
```

Comment:
Render `@note` items as a markdown list because LuaLS doesn't appear to support `@note` items, or at least our hover doc doesn't render them specially. This is easy to change later if our hover doc or LuaLS gains support.